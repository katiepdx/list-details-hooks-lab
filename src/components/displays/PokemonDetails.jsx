// list all pokemon details - pokemon, url_image, hp, attack, defense, ability_1,

import React from 'react'
import propTypes from 'prop-types'

export default function PokemonDetails({ pokemon, hp, attack, defense, ability_1, url_image }) {
  return (
    <div>
      PokemonDetails page
      <p>Pokemon: {pokemon}</p>
      <p>HP: {hp}</p>
      <p>Attack: {attack}</p>
      <p>Defense: {defense}</p>
      <p>Ability: {ability_1}</p>
      <p>Image: {url_image}</p>
    </div>
  )
}

PokemonDetails.propType = {
  pokemon: propTypes.string.isRequired,
  hp: propTypes.number.isRequired,
  attack: propTypes.number.isRequired,
  defense: propTypes.number.isRequired,
  ability_1: propTypes.string.isRequired,
  url_image: propTypes.string.isRequired
}
