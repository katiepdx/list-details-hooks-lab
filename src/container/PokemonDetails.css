div {
  align-items: center;
  justify-content: center;
  text-align: center;
  font-size: 1.2rem;
}
