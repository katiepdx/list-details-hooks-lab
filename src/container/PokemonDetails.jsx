// list all pokemon details - pokemon, url_image, hp, attack, defense, ability_1,

import React, { useState, useEffect } from 'react'
import propTypes from 'prop-types'
import { getByPokemonId } from '../services/api-fetch'

export default function PokemonDetails({ match }) {
  // loading state - default is true
  const [loading, setLoading] = useState(true)

  // pokemon data state - initialize as an empty array
  const [pokemonData, setPokemonData] = useState([])

  // useEffect (like component did mount) for pokemon detail page.
  useEffect(async () => {
    // get id from params for api fetch
    const pokemonId = match.params.pokemonId;

    // make fetch to api for a specific pokemon using the id 
    const pokemonData = await getByPokemonId(pokemonId)

    console.log(pokemonData, 'DETAILS PAGE')

    // set pokemonData to state using setPokemonData
    setPokemonData(pokemonData)

    // update loading state to false now that we have the pokemonData
    setLoading(false)

  }, [])

  if (loading) return (<p data-testid="loading">Your content is loading...</p>)


  return (
    <div>
      <h1>You chose {pokemonData.pokemon}!</h1>
      <img src={pokemonData.url_image} />
      <p>Pokemon: {pokemonData.pokemon}</p>
      <p>HP: {pokemonData.hp}</p>
      <p>Attack: {pokemonData.attack}</p>
      <p>Defense: {pokemonData.defense}</p>
      <p>Ability: {pokemonData.ability_1}</p>
    </div>
  )
}

PokemonDetails.propType = {
  match: propTypes.object.isRequired
}
