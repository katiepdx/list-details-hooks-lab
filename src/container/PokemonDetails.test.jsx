// test that the details for one pokemon renders to the page

import React from 'react'
import { render } from '@testing-library/react'
import PokemonDetails from './PokemonDetails'

describe('Tests PokemonDetails component', () => {
  it('should render the details for one pokemon to the page', () => {
    const { asFragment } = render(<PokemonDetails
      pokemon="Pokemon Name"
      hp={5}
      attack={200}
      defense={100}
      ability="overgrow"
      url_image="http://assets.pokemon.com/assets/cms2/img/pokedex/full/003.png"
      key="Pokemon Name"
    />)

    expect(asFragment()).toMatchSnapshot();
  })
})
