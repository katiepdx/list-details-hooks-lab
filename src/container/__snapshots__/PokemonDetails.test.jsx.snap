// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests PokemonDetails component should render the details for one pokemon to the page 1`] = `
<div
  data-testid="display"
>
  <h1>
    You chose 
    !
  </h1>
  <img />
  <p>
    Pokemon: 
  </p>
  <p>
    HP: 
  </p>
  <p>
    Attack: 
  </p>
  <p>
    Defense: 
  </p>
  <p>
    Ability: 
  </p>
</div>
`;
